src/WebApi/End2endTests/QuestionSetTests/DeleteQuestionSetTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Domain.Models;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using WebApi;
+using WebApi.UseCases.v1.QuestionSet.DeleteQuestionSet;
+using Xunit;
+
+namespace End2EndTests.QuestionSetTests
+{
+    public class DeleteQuestionSetTests : E2ETestsBase, IClassFixture<MyWebApplicationFactory<Startup>>
+    {
+        private readonly Func<int, string> _url = (int id) => $"/api/v1/QuestionSet/{id}";
+
+        public DeleteQuestionSetTests(MyWebApplicationFactory<Startup> factory) : base(factory)
+        {
+        }
+
+        private async Task<StandardResponse<QuestionSetModel>> EndpointCall(int id) => await _wrapper.DeleteAsync<QuestionSetModel>(_url(id));
+
+        [Fact]
+        public async Task Delete_ValidRequest_ReturnsNoContent()
+        {
+            var result = await EndpointCall(1);
+
+            Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
+        }
+
+        [Fact]
+        public async Task Delete_InvalidRequest_ReturnsNotFound()
```

Comment:
I am worried that with time this practice of calling tests invalid requests, without an obvious statement why it's invalid will bite us 😄 . For example, I didn't know that there is no item with ID 0.  Without that knowledge, it's hard to grasp. 

No changes are needed but let's think about the solution for the future. 
@TomasNiemczyk98 @petrspelos @TerezaSkutova

Reply:
Agreed. Overall unit tests at this moment are not in a great state (no abstraction, badly described). Right now I am personally not sure how to improve them so maybe we could create a new issue for someone to look into that?
@PlesnikJakub

Reply:
This is more commonly done by introduction of constants, helper methods, and descriptive test names.
In this case, maybe something like `Delete_NonExistentId_ReturnsNotFound`. 🤔 

I believe this can be further explained through the use of constants, for example:

```cs
const int NonExistentId = 0;

// ...
```

If multiple tests utilize a combination of valid and invalid IDs, this can even be defined in the test class itself.

What do you guys think? 🤔